community/kernel/src/main/java/org/neo4j/unsafe/batchinsert/internal/BatchInserterImpl.java

```diff
@@ -248,11 +245,10 @@ public BatchInserterImpl( final File storeDir, final FileSystemAbstraction fileS
         boolean dump = config.get( GraphDatabaseSettings.dump_configuration );
         this.idGeneratorFactory = new DefaultIdGeneratorFactory( fileSystem );
 
-        recordFormats = RecordFormatSelector.autoSelectFormat( config, logService );
-        maxNodeId = recordFormats.node().getMaxId();
-
         StoreFactory sf = new StoreFactory( this.storeDir, config, idGeneratorFactory, pageCache, fileSystem,
-                recordFormats, logService.getInternalLogProvider() );
+                logService.getInternalLogProvider() );
+
+        maxNodeId = sf.getRecordFormats().node().getMaxId();
```

Comment:
In this example storeFactory is used as record format provider, but it should not be; it's not a storeFactory responsibility